packages/flutter_test/lib/src/finders.dart

```diff
@@ -469,3 +483,24 @@ class _ElementPredicateFinder extends MatchFinder {
     return predicate(candidate);
   }
 }
+
+class _DescendantFinder extends Finder {
+  _DescendantFinder(this.ancestor, this.descendant, { bool skipOffstage: true }) : super(skipOffstage: skipOffstage);
+
+  final Finder ancestor;
+  final Finder descendant;
+
+  @override
+  String get description => '${ancestor.description} has descendant(s) with ${descendant.description} ';
+
+  @override
+  Iterable<Element> apply(Iterable<Element> candidates) {
+    return candidates.where((Element element) => descendant.evaluate().contains(element));
+  }
+
+  @override
+  Iterable<Element> get _allElements {
```

Comment:
The reason why I override the method here is because I only care about a sub widget tree (or several sub trees) here instead of the whole widget tree. So traverse the whole tree to get all elements is a waste. Would you mind if I change the visibility of this method to protected so I can legitimately override it?